main.py

```diff
@@ -53,7 +53,131 @@ def handle_arguments(cl_arguments):
+    use of pretrained target specific module parameters, then this list will 
+    cosnist of all the task names so that we avoid loading the 
+    pretrained parameters. Else, it will be an empty list.
+    strict: bool, indicates whether to exit if the loaded model has some expected
+    parameters missing
+
+    """
+    if not args.do_target_task_training:
+        log.info("In strict mode because do_target_task_training is off. "
+                 "Will crash if any tasks are missing from the checkpoint.")
+        strict = True
+    else:
+        strict = False
+
+    if args.do_target_task_training and not args.allow_reuse_of_pretraining_parameters:
+        # If we're training models for evaluation, which is always done from scratch with a fresh
+        # optimizer, we shouldn't load parameters for those models.
+        # Usually, there won't be trained parameters to skip, but this can happen if a run is killed
+        # during the do_target_task_training phase.
+        task_names_to_avoid_loading = [task.name for task in target_tasks]
+    else:
+        task_names_to_avoid_loading = []
+
+    if not args.load_eval_checkpoint == "none":
+        # This is to load a particular eval checkpoint.
+        log.info("Loading existing model from %s...", args.load_eval_checkpoint)
+        load_model_state(model, args.load_eval_checkpoint,
+                         args.cuda, task_names_to_avoid_loading, strict=strict)
+    else:
+        # Look for eval checkpoints (available only if we're restoring from a run that already
+        # finished), then look for training checkpoints.
+
+        if args.transfer_paradigm == "finetune":
+            # Save model so we have a checkpoint to go back to after each task-specific finetune.
+            model_state = model.state_dict()
+            model_path = os.path.join(args.run_dir, "model_state_untrained_prefinetune.th")
+            torch.save(model_state, model_path)
+
+        best_path = get_best_checkpoint_path(args.run_dir)
+        if best_path:
+            load_model_state(model, best_path, args.cuda, task_names_to_avoid_loading,
+                             strict=strict)
+        else:
+            assert_for_log(args.allow_untrained_encoder_parameters,
+                           "No best checkpoint found to evaluate.")
+            log.warning("Evaluating untrained encoder parameters!")
+    return task_names_to_avoid_loading, strict
+
+def check_configeration(args, pretrain_tasks, target_tasks):
+    """
+    Checks configerations for any obvious logical flows
+    and that necessary parameters are set for each step - 
+    throws asserts and exits if found.
+
+    Parameters
+    ----------------
+    args: Params object
+    pretrain_tasks: list of pretraining Task objects
+    target_tasks: list of target task training Task objects
+
+    Returns
+    ----------------
+    None
+    """
+    steps_log = []
```

Comment:
Why not just log as you go?

Otherwise, prefer using [`io.StringIO`](https://docs.python.org/3/library/io.html) to build up a string incrementally.